jupyterlab/extension.py

```diff
@@ -211,6 +211,7 @@ def load_jupyter_server_extension(nbapp):
         page_config['hubPrefix'] = nbapp.hub_prefix
         page_config['hubHost'] = nbapp.hub_host
         page_config['hubUser'] = nbapp.user
+        page_config['hubServerName'] = nbapp.server_name
```

Comment:
I think we need a guard here for `hasattr(nbapp, 'server_name')`

Reply:
My understanding was that the default server name is the empty string (as @rcthomas said). If URLExt.join is smart and ignores empty strings as a no-op then presumably such a switch would be unnecessary. That being said I am not sure.

Reply:
See here: https://github.com/jupyterhub/jupyterhub/blob/master/jupyterhub/singleuser.py#L449

Reply:
If the issue is that the server wouldn't have that attribute if it was just a single person running JupyterLab on their laptop, then why aren't guards `hasattr(nbapp, 'hub_prefix')`, `hasattr(nbapp, 'hub_host')`, and `hasattr(nbapp, 'user')` also necessary? I'm a little confused.

Reply:
What I meant is to check whether the property exists as a way to detect whether we are using jupyterhub 1.0 or not.

Reply:
Alright then it seems like it might not be necessary since the `server_name` attribute existed pre-1.0. https://github.com/jupyterhub/jupyterhub/pull/2089

I think it was called `name` before my PR (although I'm not 100%), but it was in master at least as early as September 2018.

Reply:
> it was in master at least as early as September 2018.

FYI, it looks like that PR was merged and released starting with 1.0 (https://github.com/jupyterhub/jupyterhub/commit/f2fa067025bf9604d58130d94dbd6265030c849e - see the tags that contain that commit)

Reply:
@jasongrout Oh oops, you're right. I'm not very good with understanding how the versioning for these things work. Just because it was merged doesn't mean it was in the current most recent release for that time.

That _does_ make sense, since the UI for named servers only showed up with 1.0 and not any earlier versions.

Now I feel dumb. I thought changing the name from `name` to `server_name` would make the code easier to understand/work with, but that seems to not have been the case. Anyway @blink1073 is definitely right about needing the `hasattr`. Sorry about that; I think I'm getting overly defensive/insecure about the changes I made/proposed.